From 83358bf31c2326ba4117defd80fb03df54220e1d Mon Sep 17 00:00:00 2001
From: matyas <matyas@4e558342-562e-0410-864c-e07659590f8c>
Date: Wed, 8 Jul 2015 20:14:16 +0000
Subject: [PATCH] osg-promote: exact route names should never be ambiguous

The added 'upcoming-prerelease' route made '-r upcoming' ambiguous,
since it can match both 'upcoming' and 'upcoming-prerelease'.  If the
given route name exactly matches one of the routes, then use that route
instead of complaining about the ambiguity.

git-svn-id: file:///p/vdt/workspace/svn/software/osg-build/trunk@21302 4e558342-562e-0410-864c-e07659590f8c
---
 osgbuild/promoter.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/osgbuild/promoter.py b/osgbuild/promoter.py
index 1382e4e..3ebd88e 100644
--- a/osgbuild/promoter.py
+++ b/osgbuild/promoter.py
@@ -704,10 +704,14 @@ def parse_cmdline_args(all_dvers, valid_routes, argv):
                 expanded_routes.extend(route.split(','))
             else:
                 expanded_routes.append(route)
         # User is allowed to specify the shortest unambiguous prefix of a route
         for route in expanded_routes:
+            if route in valid_routes.keys():
+                # exact match
+                matched_routes.append(route)
+                continue
             matching_routes = [x for x in valid_routes.keys() if x.startswith(route)]
             if len(matching_routes) > 1:
                 parser.error("Ambiguous route %r. Matching routes are: %s" % (route, ", ".join(matching_routes)))
             elif not matching_routes:
                 parser.error("Invalid route %r. Valid routes are: %s" % (route, ", ".join(valid_routes.keys())))
-- 
1.8.3.1

